templates/filebeat.yml.j2

```diff
@@ -13,10 +13,14 @@ output:
     # Array of hosts to connect to.
     hosts: {{ filebeat_output_elasticsearch_hosts | to_json }}
 
-    # Optional protocol and basic auth credentials. These are deprecated.
-    #protocol: "https"
-    #username: "admin"
-    #password: "s3cr3t"
+    # Optional auth via API Key or username/password.
+    # The options are mutually exclusive and api_key takes the precedence.
+    {% if 'api_key' in filebeat_output_elasticsearch_auth %}
+    api_key: "{{ filebeat_output_elasticsearch_auth.api_key }}"
+    {% elif 'username' in filebeat_output_elasticsearch_auth %}
+    username: "{{ filebeat_output_elasticsearch_auth.username }}"
```

Comment:
The indentation on the username field is wrong and will come out doubly indented. I think you need to use `{%-` in the elif statement.

Reply:
Indeed, i faced this issue myself. I'll update the PR!